investigator/investigator/common.py

```diff
@@ -31,26 +30,20 @@
 from thoth.sourcemanagement.sourcemanagement import SourceManagement
 from thoth.sourcemanagement.enums import ServiceType
 
-_LOG_SOLVER = os.environ.get("THOTH_LOG_SOLVER") == "DEBUG"
+from .configuration import Configuration
 
 _LOGGER = logging.getLogger(__name__)
 
-_LOG_REVSOLVER = os.environ.get("THOTH_LOG_REVSOLVER") == "DEBUG"
-GITHUB_PRIVATE_TOKEN = os.getenv("THOTH_GITHUB_PRIVATE_TOKEN")
-GITLAB_PRIVATE_TOKEN = os.getenv("THOTH_GITLAB_PRIVATE_TOKEN")
-SLEEP_TIME = int(os.getenv("ARGO_PENDING_SLEEP_TIME", 2))
-_PENDING_WORKFLOW_LIMIT = os.getenv("ARGO_PENDING_WORKFLOW_LIMIT", None)
 
-
-async def wait_for_limit(openshift: OpenShift):
+async def wait_for_limit(openshift: OpenShift, workflow_namespace: str):
     """Wait for pending workflow limit."""
     total_pending = inf
-    if _PENDING_WORKFLOW_LIMIT is None:
+    if Configuration.PENDING_WORKFLOW_LIMIT is None:
         return
-    limit = int(_PENDING_WORKFLOW_LIMIT)
+    limit = int(Configuration.PENDING_WORKFLOW_LIMIT)
     while total_pending > limit:
-        await sleep(SLEEP_TIME)
-        total_pending = openshift.workflow_manager.get_pending_workflows()
+        await sleep(Configuration.SLEEP_TIME)
+        total_pending = openshift.workflow_manager.get_pending_workflows(workflow_namespace=workflow_namespace)
```

Comment:
I realized that here we should call `get_pending_workflows` and set it to total_pending before waiting, that way we don't sleep unnecessarily

Reply:
so I will invert line 45 and 46

Reply:
Order is fine, it's the instantiation of total_pending, it should happen right before the while loop.

```python
total_pending = get_pending()
while total_pending > limit:
  await sleep()
  total_pending = get_pending()
```

Reply:
ah right! Thanks!!

Reply:
Added